core/drivers/crypto/caam/caam_ctrl.c

```diff
@@ -130,6 +131,13 @@ static TEE_Result crypto_driver_init(void)
 		goto exit_init;
 	}
 
+	/* Initialize the Manufacturing Protection Module */
+	retstatus = caam_mp_init(jrcfg.base);
+	if (retstatus != CAAM_NO_ERROR && retstatus != CAAM_NOT_SUPPORTED) {
```

Comment:
Yes, that looks weird I agree, but some early board revision does not support the feature (even though they should). So, to avoid the `panic()`, `CAAM_NOT_SUPPORTED` is considered OK. Any MP feature will fail anyway on these boards